### PR TITLE
Omit dependencyManagement module from direct dependency

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

As `dependencyManagement` is directly added as a project dependency to all projects, it is listed in the POM file and Gradle module metadata. Consequently, the `dependencyManagement` module is manually removed from the POM dependencies. However, there is no extension point to customize the Gradle module metadata file.

It would be a easier way to remove `dependencyManagement` from the project dependency fundamentally rather than removing later when publishing.

Modifications:

- Create `internal` configuration and add `dependencyManagement` dependency to it.
- `internal` configuration is set as the parent configuration of `*Classpath`, shaded, and Proguard configurations.
- Removed the legacy configuration format `dependencies.yaml` to remove maintenance costs

Result:

- `gradle-scripts` correctly produces Gradle module metadata if no `trim` or `shade` flags are set
- ⚠️ `dependencies.yaml` is no longer supported. Please use `dependencies.toml` instead.

### CODE REVIEW

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Comment by minwoox** [comment]
@ikhoon 👍 👍 👍

File: lib/java-shade.gradle [hunk 1]
```diff
@@ -231,6 +233,7 @@ configure(relocatedProjects) {
         canBeResolved = true
         canBeConsumed = false
         extendsFrom(project.configurations.shadedJarTestImplementation)
+        extendsFrom(project.configurations.depedencyManagement)
```
[the remark this answers is not in the document]
Comment (ikhoon) [reply lib/java-shade.gradle:236]: 😱😱😱

[the remark this answers is not in the document]
Comment (minwoox) [reply lib/java-shade.gradle:236]: 😱😱😱

**Pull request merged**